tabular/src/autogluon/tabular/models/tabular_nn/torch/tabular_torch_dataset.py

```diff
@@ -122,17 +122,49 @@ def __init__(self, processed_array, feature_arraycol_map, feature_type_map, prob
         self.num_categories_per_embed_feature = None
         self.num_categories_per_embedfeature = self.getNumCategoriesEmbeddings()
 
-    def __getitem__(self, idx):
-        output_dict = {}
-        if self.has_vector_features():
-            output_dict['vector'] = self.data_list[self.vectordata_index][idx]
-        if self.num_embed_features() > 0:
-            output_dict['embed'] = []
-            for i in self.embed_indices:
-                output_dict['embed'].append(self.data_list[i][idx])
-        if self.label_index is not None:
-            output_dict['label'] = self.data_list[self.label_index][idx]
-        return output_dict
+        self.has_vector_features = self.vectordata_index is not None
+        self.has_embed_features = len(self.feature_groups['embed']) > 0
+
+    def __iter__(self):
+        """
+        Iterate through the iterable dataset, and return a subsample of it.
+
+        This overrides the `__iter__` function in IterableDataset.
+        This is typically useful when we are using :class:`torch.utils.data.DataLoader` to
+        load the dataset.
+
+        Returns a tuple containing (vector_features, embed_features, label).
+        The length of the tuple depends on `has_vector_features` and `has_embed_feautures` attribute.
+        """
+        idxarray = np.arange(self.num_examples)
+        if self.shuffle:
+            np.random.shuffle(idxarray)
+        indices = range(0, self.num_examples, self.batch_size)
+        for idx_start in indices:
+            # Drop last batch
+            if self.drop_last and (idx_start + self.batch_size) > self.num_examples:
+                break
```

Comment:
Were we dropping last before this PR?

Reply:
This depends on the drop_last argument in the data loader.

Reply:
I think I found the answer, we were, if you look at deleted code lines

Reply:
We won't drop_last or shuffle when is_test==True, we were dropping last and shuffle the dataset for training. The behavior is maintained to be consistent.